packages/transactions/tests/builder.test.ts

```diff
@@ -2359,3 +2370,279 @@ test.each([
+      const signerKeys = signing.map(pk => privateKeyToString(pk)); // also works out-of-order
+
+      const tx = await makeSTXTokenTransfer({
+        recipient: 'STB44HYPYAT2BB2QE513NSP81HTMYWBJP02HPGK6',
+        amount: 12_345n,
+        fee: 1_000n,
+        nonce: 2n,
+        network: 'testnet',
+        anchorMode: AnchorMode.Any,
+        numSignatures: required,
+        publicKeys,
+        signerKeys,
+        useNonSequentialMultiSig: true,
+      });
+
+      expect(() => tx.verifyOrigin()).not.toThrow();
+    });
+
+    test.each(CASES)('works when signing separately', async ({ signers, signing, required }) => {
+      const publicKeys = signers.map(pk => publicKeyToString(getPublicKey(pk)));
+      const signerKeys = signing.map(pk => privateKeyToString(pk));
+
+      const txInitial = await makeSTXTokenTransfer({
+        recipient: 'STB44HYPYAT2BB2QE513NSP81HTMYWBJP02HPGK6',
+        amount: 12_345n,
+        fee: 1_000n,
+        nonce: 2n,
+        network: 'testnet',
+        anchorMode: AnchorMode.Any,
+        numSignatures: required,
+        publicKeys,
+        signerKeys: [], // empty at first (e.g. created by a web interface)
+        useNonSequentialMultiSig: true,
+      });
+
+      let serialized = transactionToHex(txInitial); // unsigned, so far (but has public key auth fields instead)
+
+      // mimic signing and serializing in between (e.g. shared group of people over the web)
+      for (const signerKey of signerKeys) {
+        // todo: this whole for loop content should be a helper function in TransactionSigner or StacksTransaction
+
+        const tx = deserializeTransaction(serialized);
+
+        // replace signers public key, with signature
+        const { nextSig } = nextSignature(
+          tx.signBegin(),
+          tx.auth.authType,
+          tx.auth.spendingCondition.fee,
+          tx.auth.spendingCondition.nonce,
+          createStacksPrivateKey(signerKey)
+        );
+
+        if (isSingleSig(tx.auth.spendingCondition)) throw 'type error';
+
+        const signerPublicKey = pubKeyfromPrivKey(signerKey).data;
+        const fieldIdx = tx.auth.spendingCondition.fields.findIndex(field => {
+          return (
+            field.contents.type === StacksMessageType.PublicKey &&
+            bytesToHex(field.contents.data) == bytesToHex(signerPublicKey)
+          );
+        });
+        tx.auth.spendingCondition.fields[fieldIdx] = createTransactionAuthField(
+          hexToBytes(signerKey).byteLength === PRIVATE_KEY_COMPRESSED_LENGTH
+            ? PubKeyEncoding.Compressed
+            : PubKeyEncoding.Uncompressed,
+          nextSig
+        );
+
+        serialized = transactionToHex(tx);
+      }
+
+      const txFinal = deserializeTransaction(serialized);
+      expect(() => txFinal.verifyOrigin()).not.toThrow();
+    });
+
+    test('non sequential multi-sig test vector', async () => {
+      // test vector generated from manual test in stacks-core
+      const EXPECTED =
```

Comment:
I assume have validated this hex string with `stacks-inspect decode-tx`?

Reply:
I just tried this, it does decode and the signatures are valid

Reply:
Thanks for checking. I didn't, but I ran the rust verify version on the instance when constructing it and parsing it from hex.